setmscver.bat

```diff
@@ -4,4 +4,7 @@ cl /nologo /EP ver.c > ver_raw.txt
 findstr /v /r /c:"^$" "ver_raw.txt" > "ver.txt"
 set /P _MSC_VER=< ver.txt
 echo set _MSC_VER=%_MSC_VER%
+if %_MSC_VER% GTR 1900 echo /std:c++17 > cflags.txt
+if %_MSC_VER% GTR 1900 echo -extern-std=c++17 > dflags.txt
+if %_MSC_VER% GTR 1900 echo string_view > add_tests.txt
```

Comment:
Would be good to remove these files for lower versions, so you don't get a stale copy when trying different versions of VS one after the other.

Reply:
I though that too, but I couldn't work out how to delete them. `del /f` errors if they don't exist, and they don't exist on the auto-tester build.

Reply:
Just put `if exist cflags.txt del /q cflags.txt` etc. before the if statements.

Reply:
DOS is so shit ;)